Update yml file

mds010 opened the pull request:
updates to yml and config for publishing to pypi

mds010 committed:
update workflow file

mds010 committed:
update distribution building command

mds010 committed:
update version number

mds010 committed:
remove file

mds010 committed:
update workflow file

harrycheon commented on pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "reachml"
-version = "0.2.0"
+version = "0.2.1.2"
```
Why .2 (instead of 0.2.1)

harrycheon committed:
ci: test publish to testpypi

- split ci workflow to: test and publish
- update pythons to include 3.11, 3.12

harrycheon committed:
fix(ci): split build and publish

- only build on 3.12
- upload to pypi directly

harrycheon committed:
fix(ci): versioning and typos

- version from tags using hatch-vcs
- fix typo in `publish.yml` workflow

harrycheon committed:
fix(ci): ignore `_version.py`
